src/main/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2.java

```diff
@@ -123,19 +123,27 @@ public SnowflakeSinkServiceV2(
   @Override
   public void startTask(String tableName, String topic, int partition) {
     String partitionChannelKey = partitionChannelKey(topic, partition);
-    LOGGER.info(
-        "Opening a channel with name:{} for table name:{}, topic:{}, partition:{}",
-        partitionChannelKey,
-        tableName,
-        topic,
-        partition);
+
     // the table should be present before opening a channel so lets do a table existence check here
     createTableIfNotExists(tableName);
+
+    createOrUpdatePartitionsToChannelCache(partitionChannelKey, tableName);
+  }
+
+  /* We will always open a new channel. This will either update the TopicPartitionChannel's streamingChannel or will create a new instance of TopicPartitionChannel. */
+  private void createOrUpdatePartitionsToChannelCache(
+      final String partitionChannelKey, final String tableName) {
+    // Always open a new channel.
```

Comment:
Have you consider the suggestion about don't close any channel during `close` and only open the channel if `partitionsToChannel.containsKey = false`?

Reply:
The only reason why I am doing a closing of client is because I want to close all underlying channels. 

consider this scenario: there is a rebalance but the task is still running, in that case, if I am trying to reopen a channel which was already opened, it will throw up because the channel name can also be deterministic right? 

What would be side effect of closing and reopening of channels?

Reply:
Actually, I just checked the close channel logic, we won't call flush if the channel is empty, so for the rebalance case, we should only flush once even we have 100 channels, what we should do is instead of calling `closeStreamingClient`, just close the channels one by one.

Reply:
Well, the closing of client internally calls markClosed() which is what each channel close does. So this is much simpler IMO..

Reply:
But I see the benefit of not closing the client. I will remove that from close() and keep it in closeAll() only.